crates/edr_provider/tests/issues/issue_570.rs

```diff
@@ -0,0 +1,100 @@
+use std::str::FromStr as _;
+
+use edr_eth::{spec::HardforkActivations, SpecId, B256};
+use edr_provider::{
+    hardhat_rpc_types::ForkConfig, test_utils::create_test_config_with_fork, time::CurrentTime,
+    MethodInvocation, NoopLogger, Provider, ProviderRequest,
+};
+use edr_test_utils::env::get_alchemy_url;
+use tokio::runtime;
+
+// `eth_debugTraceTransaction` should return a helpful error message if there is
+// a transaction in the block whose type is not supported.
+// https://github.com/NomicFoundation/edr/issues/570
+#[tokio::test(flavor = "multi_thread")]
+async fn issue_570_error_message() -> anyhow::Result<()> {
+    let logger = Box::new(NoopLogger);
+    let subscriber = Box::new(|_event| {});
+
+    let mut config = create_test_config_with_fork(Some(ForkConfig {
+        json_rpc_url: get_alchemy_url().replace("eth-mainnet", "base-sepolia"),
+        block_number: Some(13_560_400),
+        http_headers: None,
+    }));
+
+    let chain_id = 84532;
```

Comment:
This can be a `const`

Reply:
Fixed in: https://github.com/NomicFoundation/edr/pull/606/commits/0e66f07f3fce6f2646fb479b538d003caf5ab95e

Reply:
Actually, really fixed it in [b11aac7](https://github.com/NomicFoundation/edr/pull/606/commits/b11aac74a3a5a53db664a29f0799acd64abec75a)